### PR TITLE
CFK-1643 - Kafka Rest Proxy samples [skip secret scan]

### PR DESCRIPTION
**Pull request by domenicbove**

### CODE REVIEW

File: general/cp-version/confluent-platform-6.2.0.yaml [hunk 1]
```diff
@@ -123,5 +123,23 @@ spec:
...
+metadata:
+  name: kafkarestproxy
+  namespace: confluent
+spec:
+  dependencies:
+    schemaRegistry:
+      url: http://schemaregistry.confluent.svc.cluster.local:8081
+  image:
+    application: confluentinc/cp-kafka-rest:6.2.0
+    init: confluentinc/confluent-init-container:2.0.0
```
Comment (domenicbove) [inline general/cp-version/confluent-platform-6.2.0.yaml:139]:
```suggestion
    application: confluentinc/cp-kafka-rest:7.0.1
    init: confluentinc/confluent-init-container:2.2.0-1
```

File: quickstart-deploy/confluent-platform.yaml [hunk 2]
```diff
@@ -87,6 +86,17 @@ spec:
...
+  name: kafkarestproxy
+  namespace: confluent
+spec:
+  replicas: 1
+  image:
+    application: confluentinc/cp-schema-registry:7.0.1
+    init: confluentinc/confluent-init-container:2.2.0-1
+---
+apiVersion: platform.confluent.io/v1beta1
+kind: KafkaRestProxy
```
Comment (domenicbove) [inline quickstart-deploy/confluent-platform.yaml:99]:
```suggestion
```

File: security/autogenerated-tls_only/confluent-platform-tls-only.yaml [hunk 3]
```diff
@@ -135,3 +135,21 @@ spec:
...
+spec:
+  replicas: 1
+  image:
+    application: confluentinc/cp-kafka-rest:7.0.1
+    init: confluentinc/confluent-init-container:2.2.0-1
+  tls:
+    autoGeneratedCerts: true
+  dependencies:
+    schemaRegistry:
+      url: http://schemaregistry.confluent.svc.cluster.local:8081
```
Comment (MosheBlumbergX) [inline security/autogenerated-tls_only/confluent-platform-tls-only.yaml:153]: shouldn't this be tls - https ?

**Review by MosheBlumbergX — CHANGES_REQUESTED** [review]

**Review by akhileshm1 — APPROVED** [review]
Good stuff @domenicbove

**Review by MosheBlumbergX — APPROVED** [review]

**Comment by domenicbove** [comment]
@rohit2b Can you review/merge?

File: hybrid/ccloud-integration/confluent-platform.yaml [hunk 4]
```diff
@@ -105,3 +105,21 @@ spec:
...
         enabled: true
+---
+apiVersion: platform.confluent.io/v1beta1
+kind: KafkaRestProxy
+metadata:
+  name: kafkarestproxy
+  namespace: confluent
+spec:
+  dependencies:
+    schemaRegistry:
```
Comment (MosheBlumbergX) [inline hybrid/ccloud-integration/confluent-platform.yaml:116]: shouldn't there be a kafka dependencies as well ?

File: hybrid/ccloud-integration/confluent-platform.yaml [hunk 5]
```diff
@@ -105,3 +105,21 @@ spec:
...
       tls:
         enabled: true
+---
+apiVersion: platform.confluent.io/v1beta1
+kind: KafkaRestProxy
+metadata:
+  name: kafkarestproxy
+  namespace: confluent
+spec:
+  dependencies:
```
Comment (domenicbove) [inline hybrid/ccloud-integration/confluent-platform.yaml:115]:
```suggestion
  dependencies:
    kafka:
      bootstrapEndpoint: <cloudKafka_url>:9092
      authentication:
        type: plain
        jaasConfig:
          secretRef: cloud-plain
      tls:
        enabled: true
        ignoreTrustStoreConfig: true 
```

File: hybrid/ccloud-integration/confluent-platform.yaml (hunk 4 above)
Comment (domenicbove) [reply hybrid/ccloud-integration/confluent-platform.yaml:125]: nice catch thanks!

**Review by rohit2b — COMMENTED** [review]
LGTM!

**Pull request merged**